crates/matrix-sdk-ui/src/timeline/sliding_sync_ext.rs

```diff
@@ -52,3 +62,211 @@ fn sliding_sync_timeline_builder(room: &SlidingSyncRoom) -> Option<TimelineBuild
         }
     }
 }
+
+/// Wrap a low-level event from a sync into a high-level EventTimelineItem,
+/// ready to be used in the message preview in a client.
+fn wrap(room: &SlidingSyncRoom, sync_event: SyncTimelineEvent) -> Option<EventTimelineItem> {
```

Comment:
Maybe it would make sense to make this a constructor of `EventImelineItem`?

Reply:
Currently it only supports `m.room.message` events, so it might be misleading to make it a constructor which might imply it would work for all event types?

Reply:
What about a constructor like `EventTimelineItem::new_from_displayable_sync_event()`? Like @poljar, I find it a bit unusual to have this (good) logic here :-).

Reply:
Done in 5369d1b50e